public/pages/workflows/new_workflow/utils.ts

```diff
@@ -138,68 +145,136 @@ export function fetchEmptyUIConfig(): WorkflowConfig {
   };
 }
 
-export function fetchSemanticSearchMetadata(): UIState {
+export function fetchSemanticSearchMetadata(version: string): UIState {
+  if (!version) {
+    throw new Error(
+      'Version parameter is required for fetchSemanticSearchMetadata'
+    );
+  }
   let baseState = fetchEmptyMetadata();
   baseState.type = WORKFLOW_TYPE.SEMANTIC_SEARCH;
-  baseState.config.ingest.enrich.processors = [new MLIngestProcessor().toObj()];
+  const isPreV219 = semver.lt(version, MINIMUM_FULL_SUPPORTED_VERSION);
+
+  baseState.config.ingest.enrich.processors = isPreV219
+    ? [
+        new TextEmbeddingIngestProcessor().toObj(),
+        new TextImageEmbeddingIngestProcessor().toObj(),
```

Comment:
For semantic search, this should just be the text embedding processor. See [semantic search documentation](https://opensearch.org/docs/latest/search-plugins/semantic-search/#step-1-create-an-ingest-pipeline) for reference

Reply:
address on line 160